### PR TITLE
Matter cleanup

### PR DESCRIPTION
**Pull request by tecimovic**
The purpose of this PR is to cleanup all the Matter helper code.

The transition process during the prolonged cut-over is like this:
   - continuously copy the .js files from Matter onto an "import" branch.
   - provide manual fixes on the matter-cleanup branch.
   - whenever a new copy comes to "import" branch, perform a merge from import branch to matter-cleanup branch.

### CODE REVIEW

File: src-electron/generator/generation-engine.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ const dbEnum = require('../../src-shared/db-enum.js')
 const env = require('../util/env')
 const templateEngine = require('./template-engine.js')
 const dbApi = require('../db/db-api.js')
+const { include } = require('underscore')
```
Comment (brdandu) [inline src-electron/generator/generation-engine.js:31]: Any reason this is named include?

File: src-electron/generator/matter/app/zap-templates/common/ChipTypesHelper.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      return 'uint64_t'
+    default:
+      return type
+  }
+}
+
+//
+// Module exports
+//
+exports.asBasicType = asBasicType
```
Comment (brdandu) [inline src-electron/generator/matter/app/zap-templates/common/ChipTypesHelper.js:54]: Do deprecation for underscore format as_basic_type?

File: src-electron/generator/matter/app/zap-templates/common/ClustersHelper.js [hunk 3]
```diff
@@ -0,0 +1,1055 @@
...
+    loadBitmaps.call(context, packageId),
+    loadStructs.call(context, packageId),
+  ]
+
+  const promises = [
+    Promise.all(loadTypes),
+    loadEndpoints.call(context),
+    // For now just always use loadClusters, because we have a bunch of things
+    // defined in our XML that are not actually part of Matter.
+    (includeAll ? loadClusters : loadClusters).call(context, packageId),
```
Comment (brdandu) [inline src-electron/generator/matter/app/zap-templates/common/ClustersHelper.js:813]: loadClusters.call(context, packageId)

**Review by brdandu — APPROVED** [review]
I am good with all of this coming into the zap repo first and we can clean it up in the next phase.

File: src-electron/generator/generation-engine.js (hunk 1 above)
Comment (tecimovic) [reply src-electron/generator/generation-engine.js:31]: No good reason. I'll remove it.

File: src-electron/generator/matter/app/zap-templates/common/ChipTypesHelper.js (hunk 2 above)
Comment (tecimovic) [reply src-electron/generator/matter/app/zap-templates/common/ChipTypesHelper.js:54]: I'm not dealing with this before we fully migrate. I keep changes isolated to just the minimum I have to do, in order to be able to regenerate Matter.
There will be a larger cleanup effort once this is merged.

File: src-electron/generator/matter/app/zap-templates/common/ClustersHelper.js (hunk 3 above)
Comment (tecimovic) [reply src-electron/generator/matter/app/zap-templates/common/ClustersHelper.js:813]: ?

**Review by paulr34 — APPROVED** [review]

**Comment by tecimovic** [comment]
@bzbarsky-apple : I disabled ASI and mandated semicolons in all generator/matter/**/*.js source files.

**Comment by tecimovic** [comment]
@bzbarsky-apple : this is now ready to be merged.

Since last time you checked:
   - the matter .js code now has semicolons.
   - all the tests.js files have been removed. Instead, there is a bunch of ciTests.json and manualTests.json files on the SDK itself. (which will be brought in with a separate PR on the Matter repo, once this merges in.)

**Pull request merged**